python/pyspark/ml/torch/distributor.py

```diff
@@ -643,13 +664,48 @@ def _setup_files(train_fn: Callable, *args: Any) -> Generator[Tuple[str, str], N
         finally:
             TorchDistributor._cleanup_files(save_dir)
 
+    @staticmethod
+    @contextmanager
+    def _setup_spark_partition_data(partition_data_iterator, input_schema_json):
+        from pyspark.sql.pandas.serializers import ArrowStreamSerializer
+        import json
+
+        if input_schema_json is None:
+            yield
+            return
+
+        save_dir = TorchDistributor._create_save_dir()
+
+        try:
+            serializer = ArrowStreamSerializer()
```

Comment:
ditto, I guess we can reuse `LocalDataToArrowConversion` to convert rows to arrow table

Reply:
https://github.com/apache/spark/pull/40724#discussion_r1161659107